station-blocks-v0/src/main/java/net/modificationstation/stationapi/mixin/block/MixinClass417.java

```diff
@@ -0,0 +1,21 @@
+package net.modificationstation.stationapi.mixin.block;
+
+import com.llamalad7.mixinextras.sugar.Local;
+import net.minecraft.class_417;
+import net.minecraft.level.Level;
+import net.modificationstation.stationapi.api.block.BlockState;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.ModifyVariable;
+
+@Mixin(class_417.class)
+public class MixinClass417 {
+	@ModifyVariable(
+		method = "method_1402(Lnet/minecraft/level/Level;)V",
+		at = @At(value = "STORE", ordinal = 2), index = 20
+	)
+	private int getStateEmittance(int original, @Local Level level, @Local(index = 10) int x, @Local(index = 15) int y, @Local(index = 11) int z) {
+		BlockState state = level.getBlockState(x, y, z);
+		return state.getBlock().getEmittance(state);
```

Comment:
This is what I was unsure about. Modern caches the emittance (or luminance) levels right in the blockstate class in a final field.

Constantly calling the block's method which in turn calls a lambda if it exists introduces some overhead compared to just retrieving a constant from the state's class directly.

But I don't know how that could be implemented. States are initialized in block's constructor, so passing something doesn't really seem possible, considering we can't add new constructors or change descriptors of the existing ones.

Reply:
That method is actually called only one time per block - in rendering when light value is captured, so there is no any huge performance impact. If you want to cache it in a blockstate it is still possible, if field will be not final it can be initiated later. Or you can use unsafe to change final field in blockstates (which is not good for design, but will work)

Reply:
Now there should be one less check, so performance should be a bit better